### PR TITLE
refactor: code review improvements to cart refactor

### PR DESCRIPTION
**Pull request by marlass**
Work based on https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/4582 PR

Notable changes:
- introduction of process loader into multicart for reloading optimization
- change in addEntry (cart load/creation stream)

### CODE REVIEW

File: projects/core/src/cart/store/effects/multi-cart.effect.ts [hunk 1]
```diff
@@ -52,34 +53,38 @@ export class MultiCartEffects {
...
-      }),
-    ])
-  );
-
-  @Effect()
-  removeCart$: Observable<CartActions.RemoveCart> = this.actions$.pipe(
-    ofType(DeprecatedCartActions.DELETE_CART),
-    map((action: DeprecatedCartActions.DeleteCart) => action.payload),
-    mergeMap(payload => [new CartActions.RemoveCart(payload.cartId)])
+    map(payload => new CartActions.CartProcessesIncrementAction(payload.cartId))
```
Comment (Platonn) [inline projects/core/src/cart/store/effects/multi-cart.effect.ts]: can we add a comment /ticket regarding the optimalization of Incrementing/Decrementing in 2.0?

Comment (marlass) [reply projects/core/src/cart/store/effects/multi-cart.effect.ts]: Added here: https://github.com/SAP/cloud-commerce-spartacus-storefront/issues/5234

File: projects/core/src/cart/facade/active-cart.service.ts [hunk 2]
```diff
@@ -133,23 +133,30 @@ export class ActiveCartService {
...
     );
   }
 
   /**
-   * Returns loaded flag (success or error)
+   * Returns true when cart is stable (not loading and not pending processes on cart)
    */
   getLoaded(): Observable<boolean> {
-    return this.cartSelector$.pipe(
-      map(cart => (cart.success || cart.error) && !cart.loading)
```
[the remark this answers is not in the document]
Comment (marlass) [reply projects/core/src/cart/facade/active-cart.service.ts]: This functions should be renamed to `isStable`. In some places `!getLoaded()` indicates loading, but it is not the case with `error/success` flags. For enonymous user without cart this function will be returning false for the whole time, which we don't want. Better naming in cart refactor for 2.0 should make this function behavior more clear.

File: projects/core/src/cart/facade/active-cart.service.ts [hunk 3]
```diff
@@ -133,23 +133,30 @@ export class ActiveCartService {
...
+    // Debounce is used here, to avoid flickering when we switch between different cart entities.
+    // For example during `addEntry` method. We might try to load current cart, so `current cart will be then active id.
+    // After load fails we might create new cart so we switch to `fresh` cart entity used when creating cart.
+    // At the end we finally switch to cart `code` for cart id. Between those switches cart `getLoaded` function should not flicker.
+    return this.activeCartId$.pipe(
+      switchMap(cartId => this.multiCartService.isStable(cartId)),
+      debounce(state => (state ? timer(0) : EMPTY)),
```
[the remark this answers is not in the document]
Comment (marlass) [reply projects/core/src/cart/facade/active-cart.service.ts]: It is, but here is the case with cart switching

**Pull request merged**